src/import/pull-job.c

```diff
@@ -101,6 +101,10 @@ void pull_job_curl_on_finished(CurlGlue *g, CURL *curl, CURLcode result) {
                 j->etag_exists = true;
                 r = 0;
                 goto finish;
+        } else if (status == 404 && endswith(j->url, "sha256")) {
+                log_info(".sha256 suffixed file not found, trying SHA256SUMS.");
+                r = 404;
+                goto finish;
```

Comment:
please don't propagate this up. Please restart the CURL job immediately with the new location, so that the users of this don't notice what is happening here.